slack_notification/slack_notification.php

```diff
@@ -84,32 +89,55 @@
     // Build an array of fields to be rendered with Slack Attachments as a table
     // attachment-style formatting:
     // https://api.slack.com/docs/attachments
-    $fields += array(
-      array(
-        'title' => 'Commit',
-        'value' => rtrim($hash),
-        'short' => 'true'
-      ),
-      array(
-        'title' => 'Commit Message',
-        'value' => $message,
-        'short' => 'false'
-      )
+    $fields[] = array(
+      'title' => 'Commit',
+      'value' => rtrim($hash),
+      'short' => 'true'
+    );
+    $fields[] = array(
+      'title' => 'Commit Message',
+      'value' => $message,
+      'short' => 'false'
     );
+    $attachment += array('pretext' => 'Code syncing');
+    break;
+
+  case 'create_cloud_development_environment':
+    // Prepare the slack payload as per:
+    // https://api.slack.com/incoming-webhooks
+    $text = $_ENV['PANTHEON_ENVIRONMENT'] . ' spinning up for';
+    $text .= $_ENV['PANTHEON_SITE_NAME'] .' by '. $_POST['user_email'] .' complete!';
+    $text .= ' <https://dashboard.pantheon.io/sites/'. PANTHEON_SITE .'#'. PANTHEON_ENVIRONMENT .'/deploys|View Dashboard>';
+    $attachment += array('pretext' => 'New environment created');
+    break;
+
+  case 'cache_clear':
+    // Prepare the slack payload as per:
+    // https://api.slack.com/incoming-webhooks
+    $text = 'Cache clear for '. $_ENV['PANTHEON_ENVIRONMENT'] . ' on ';
+    $text .= $_ENV['PANTHEON_SITE_NAME'] .' by '. $_POST['user_email'] .' complete!';
+    $text .= ' <https://dashboard.pantheon.io/sites/'. PANTHEON_SITE .'#'. PANTHEON_ENVIRONMENT .'/deploys|View Dashboard>';
+    $attachment += array('pretext' => 'Caches cleared');
     break;
 
   default:
     $text = $_POST['qs_description'];
     break;
 }
 
-$attachment = array(
+// Add the prepared $text and $fields arrays
+$attachment += array(
   'fallback' => $text,
-  'pretext' => 'Deploying :rocket:',
-  'color' => $pantheon_yellow, // Can either be one of 'good', 'warning', 'danger', or any hex color code
   'fields' => $fields
 );
 
+// Add pretext default if not already defined
+if (empty($attachment->pretext)) {
+  $attachment += array(
+    'pretext' => 'Pantheon system update :rocket:'
+  );
+}
+
 _slack_notification($secrets['slack_url'], $secrets['slack_channel'], $secrets['slack_username'], $text, $attachment, $secrets['always_show_text']);
```

Comment:
Personally I think that even if it wasn't reading from the secrets file, I would probably just rework into a local config.json file that I would store parallel to the /private/scripts. 

Much easier for us to keep scripts unified across all our clients with consistent inputs & outputs that way. This is especially attractive to us as we work with multiple agencies at times and will be doing some work to make slack notifications loop through multiple slack configs and channels.

Either way if you want to take the quicksilver-examples in a different direction that is cool, I just wanted to contribute where I saw this discrepancy!